modules/create_timepoint/php/ajaxtimepoint.class.inc

```diff
@@ -0,0 +1,280 @@
+            case 'GET':
+                return $this->_handleGET($request);
+            case 'POST':
+                return $this->_handlePOST($request);
+            default:
+                return new \LORIS\Http\Response\MethodNotAllowed(
+                    $this->allowedMethods()
+                );
+        }
+    }
+    /**
+     * Initialize setup, the extra values for the
+     * create timepoint form. (psc & errors)
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values = $request->getQueryParams();
+        // Verify identifier from Post.
+        $values['identifier'] = isset($values['identifier']) ?
+            intval($values['identifier']) : null;
+        // User permission verification.
+        $values = $this->_verifyPermissions($request, $values);
+        if (isset($values['error'])) {
+            switch ($values['error']) {
+                case 'BadRequest':
+                    return new \LORIS\Http\Response\BadRequest(
+                        'Failed to retrieve data for create timepoint.'
+                    );
+                case 'Forbidden':
+                    return new \LORIS\Http\Response\Forbidden(
+                        'Forbidden'
+                    );
+                default:
+                    break;
+            }
+        }
+        // Setup variables
+        $errors         = array(); // form errors.
+        $config         = \NDB_Config::singleton();
+        $candidate      = \Candidate::singleton($values['identifier']);
+        $allSubprojects = \Utility::getSubprojectList();
+        // All subprojects from config file (error).
+        if (empty($allSubprojects)) {
+            $errors['subprojectID'] = 'No subprojects have been defined 
+            for this study. If you are an administrator, please use the 
+            Configuration module to add new subprojects.';
+        }
+        // List of valid subprojects for a given project
+        $subprojList = $candidate->getValidSubprojects();
+        if (empty($subprojList)) {
+            $this->form->errors['subprojectID'] = 'No subprojects have been 
+            defined for the project this candidate is affiliated with. 
+            If you are an administrator, please use the Configuration module to 
+            add new subprojects and associate them with projects.';
+        }
+        $values['subproject'] = array_intersect_key(
+            \Utility::getSubprojectList(),
+            $candidate->getValidSubprojects()
+        );
+        // Retrieve visit labels.
+        $visit_options      = array();
+        $visitLabelSettings = $config->getSetting('visitLabel');
+        foreach (
+            \Utility::toArray($visitLabelSettings) as $visitLabel
```

Comment:
`Utility::toArray()` is deprecated and shouldn't be used. You can use `associativeToNumericArray` instead.